### ISSUE TITLE
unsupported OVS_ACTION_ATTR_SET attribute 0  on coreOS 877.1.0

### ISSUE THREAD

**Issue by enxebre** [weaveworks/weave#1854]
Hi,
I am having ``` unsupported OVS_ACTION_ATTR_SET attribute 0  on coreOS 877.1.0 ```
when ``` weave launch```
I'm guessing related to openvswitch changes on linux 4.3 https://lwn.net/Articles/662714/ 

weave version 1.4.1
coreOS 877.1.0

Thanks

**Comment by bboreham** [comment]
Interesting, I don't get that when I try.

Could you supply details of the commands used to launch, plus full logs from `docker logs weave`?

**Comment by bboreham** [comment]
On further testing, I _do_ get something similar if I stop then re-launch weave:

````
core@ip-172-31-46-228 ~ $ ./weave launch-router 172.31.46.229
2fb3a61b029e39165a959b09319f9639de2c36d61862121ed4b29951fedf9c2b
core@ip-172-31-46-228 ~ $ ./weave stop-router
core@ip-172-31-46-228 ~ $ ./weave launch-router 172.31.46.229
The weave container has died. Consult the container logs for further details.
core@ip-172-31-46-228 ~ $ docker logs weave
INFO: 2016/01/05 16:46:35.372920 Command line options: map[ipalloc-range:10.32.0.0/12 name:1a:90:49:e6:2e:70 nickname:ip-172-31-46-228.eu-west-1.compute.internal port:6783 datapath:datapath dns-effective-listen-address:172.17.0.1 dns-listen-address:172.17.0.1:53 http-addr:127.0.0.1:6784]
INFO: 2016/01/05 16:46:35.372997 Command line peers: [172.31.46.229]
FATA: 2016/01/05 16:46:35.392626 unsupported OVS_ACTION_ATTR_SET attribute 4
````

**Comment by bboreham** [comment]
In this situation, `weave reset` then `weave launch` lets it work again, for me.

**Comment by bboreham** [comment]
Seems similar to https://github.com/weaveworks/go-odp/pull/4

**Comment by enxebre** [comment]
@bboreham weave reset then weave launch works
Thanks

**Comment by dpw** [comment]
Might be of interest that I couldn't reproduce this in my VM running kernel 4.4rc8.  I had to reboot it into 4.3 to reproduce.

So I wonder if this is not due to new _CT_ attributes in 4.3 as I had assumed, but rather is related to this: http://thread.gmane.org/gmane.network.openvswitch.devel/56655

**Comment by bboreham** [comment]
Seems to have been introduced by https://github.com/torvalds/linux/commit/fc4099f17240767554ff3a73977acb78ef615404

> While transitioning to netdev based vport we broke OVS feature [...]

and fixing that caused this breakage.

(thanks for info @rade)

**Comment by dpw** [comment]
See comments on weaveworks/go-odp#4.  Should be fixed now with an updated go-odp.

**Comment by bboreham** [comment]
@dpw I think you might have meant weaveworks/go-odp#5.  Commits are https://github.com/weaveworks/go-odp/commit/5d06488c83bbfb806e44e221bc9a5d202a634e23 and https://github.com/weaveworks/go-odp/commit/c794b8c293b0b5cfbefaa7f9d187219afb9b3d7d, thanks!

**Comment by awh** [comment]
Fixed by weaveworks/go-odp@c794b8c.
